ui/src/duck/designation.ts

```diff
@@ -0,0 +1,42 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+import { Designation, fetch } from 'src/entities/Designation';
+
+export interface DesignationState {
+  error?: string;
+  isLoading: boolean;
+  designation: Designation[];
+}
+
+export const fetchDesignation = createAsyncThunk<
+  Designation[],
+undefined,
+{ rejectValue: Error; state: DesignationState }
+  >('designations/fetch', fetch);
```

Comment:
Let's start following a convention of naming fetch action as `fetchAll` and `fetchOne`.